File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> ans = new ArrayList<>();
        
        for(int i = 1; i <= numRows; i++) {
            ans.add(generateRow(i));
        }
        return ans;
    }
    
    private List<Integer> generateRow(int row) {
        List<Integer> rows = new ArrayList<>();
        int first = 1;
        rows.add(first);
        for(int i = 1; i < row; i++) {
            first = first * (row - i);
            first = first / i;
            rows.add(first);
        }
        
        return rows;
    }
}

File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows):
        result = []
        
        for row_number in range(1, numRows + 1):
            result.append(self.generate_row(row_number))
        
        return result
    
    def generate_row(self, row_number):
        row = []
        element = 1
        row.append(element)
        
        for i in range(1, row_number):
            element = element * (row_number - i)
            element = element // i
            row.append(element)
        
        return row
